src/app/dialogs/edit-poll-dialog/edit-poll-dialog.component.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright © Live-Poll 2020-2021. All rights reserved
+ */
+
+import {Component, EventEmitter, Input, OnInit, Output} from '@angular/core';
+import {Poll} from '../../model/poll';
+import {HttpClient, HttpHeaders, HttpResponse} from '@angular/common/http';
+import {environment as env} from '../../../environments/environment';
+import {User} from '../../model/user';
+import {CommonToolsService} from '../../service/common-tools.service';
+
+@Component({
+  selector: 'app-edit-poll-dialog',
+  templateUrl: './edit-poll-dialog.component.html',
+  styleUrls: ['./edit-poll-dialog.component.sass']
+})
+export class EditPollDialogComponent implements OnInit {
+
+  // Event Emitters
+  @Input() poll: Poll;
+  @Input() userData: User;
+  @Input() isVisible: boolean;
+  @Output() onClose = new EventEmitter<boolean>(); // true = success; false = cancel
+
+  // Variables
+  loading = false;
+  name = '';
+  startDate = new Date();
+  endDate = new Date();
+
+  /**
+   * Initialize the component
+   * @param http Injected http client
+   * @param tools Injected ToolsService
+   */
+  constructor(
+    private http: HttpClient,
+    private tools: CommonToolsService
+  ) {}
+
+  ngOnInit(): void {
+    this.name = this.poll.name;
+    this.startDate = this.poll.startDate;
+    this.endDate = this.poll.endDate;
+  }
+
+  updatePoll(): void {
+    this.loading = true;
+    const name = this.name;
+    const startDate = this.startDate;
+    const endDate = this.endDate;
+
+    // Build header, body and options
+    const header = new HttpHeaders().set('Content-Type', 'application/json');
+    const options: any = { header, responseType: 'text', observe: 'response', withCredentials: true };
+    const body = { name, startDate, endDate };
+    // Send request
+    this.http.put<string>(env.apiBaseUrl + '/users/' + this.userData.id + '/poll/' + this.poll.id, body, options)
```

Comment:
api has changed

Reply:
The intention of this pr is to preserve the working version with the old api endpoints and to draw a line between the changes from the old semester and the new ones.
I don't now if I even want to squash all those dependabot commits from above. What do you think?

Reply:
>The intention of this pr is to preserve the working version with the old api endpoints and to draw a line between the changes from the old semester and the new ones.

Ah ok.
<br>

>I don't now if I even want to squash all those dependabot commits from above. 

Wait, do you want to squash or not? I mean squashing would make sense here, so we wouldn't have all those depandabots commits cluttering up the history. The downside would be you losing the commits, right? Do what feels better, I'd be ok with a rebase as well so that your individual commits are preserved and our teacher can see you coding ;-)